### ISSUE TITLE
Product Shops Out of Stock

### ISSUE THREAD

**Issue by bbihun12** [facebookincubator/facebook-for-woocommerce#523]
Hello, I just uploaded the latest version of the Plugin 1.8.7 and synced 30 products from my Woocommerce. Each product has 3 variations, Top Size, Bottom Size, Bottom Coverage.  

Each product I go to says out of stock and user cannot purchase the item. I have the Stock Status set for each product to In Stock, but I don't manage stock.

Why is it showing Out of Stock in the Facebook shop?

Woocommerce: https://islaraeswim.com/shop/
Facebook Page: https://www.facebook.com/islaraeswim/shop/

Thank you

**Comment by tro127** [comment]
I too am using version 1.8.7 and all products are showing as out of stock, any help?

Woocommerce: https://jecodesigns.co.uk/shop/
Facebook Page: https://www.facebook.com/Jecodesigns/shop/

Thanks

**Comment by FlairTrade** [comment]
I'm seeing the same issue with latest release or some other change on Facebook's side.
WordPress version 4.9.6
WooCommerce version 3.4.1
plugin version 1.8.7

The majority of my over 700 items currently in the catalog are showing this OUT OF STOCK state in the Facebook shop even though the Catalog availability state on every item I have is In Stock. I've included screenshots of only one item as an example.

https://www.flairtradeconsign.com/product/backpack-purse/
https://www.facebook.com/pg/flairtradeconsign/shop/
The catalog is in sync with WooCommerce, I have not tried to reset the plugin at this point as it appears to be a broader issue.

![catalog view](https://user-images.githubusercontent.com/35601034/40939194-f7cc9812-6811-11e8-9414-3835a9de6925.png)
![facebook shop view](https://user-images.githubusercontent.com/35601034/40939197-faaf7e32-6811-11e8-9261-559360e2713a.png)

**Comment by mengyingdu** [comment]
Hi @FlairTrade @bbihun12 @tro127 ,

This is caused by a recent change from Facebook shop team. The fixed is landed now. However, for the existing products with this problem, please expect 1-2 days then resync for a fix.

Sorry for this inconvenience.

**Comment by dmitridr** [comment]
Can confirm, this is not a bug in the plugin but a bug in all FB shops everywhere. 2 days is the current ETA for a fix from the shops team.

**Comment by tro127** [comment]
Thanks for the quick response

**Comment by FlairTrade** [comment]
Not shooting the messengers, but how frustrating to constantly deal with bugs like this and then be provided an a 48 hour ETA for a fix. I realize this is a "free" service but our traffic does drive significant value for Facebook. You'd think there would be a bit more love and urgency for the small business community... Thanks for the response, acknowledgement and insight @mengyingdu @dmitridr

**Comment by dmitridr** [comment]
Update : I'm told the fix is live, but the data migration to fix existing shops is in progress. If you don't want to wait for the migration, you may have some success in getting rid of the bug now if you do a `Force Resync`.

**Comment by FlairTrade** [comment]
@dmitridr thanks for the update. I can confirm doing a Force Resync fixes newly added items but not existing at this point. It looks like the only options maybe to wait for the official data migration or possibly reset which isn't ideal with existing tagged product.

**Comment by sfconcas** [comment]
I have done a force resync but didn't really help. I guess I just need to wait

**Comment by dmitridr** [comment]
Thanks for trying that. If a force resync doesn't work, you can try a full reset of the plugin, via `Reconfigure > Advanced Options > Delete`. I'll check with our shops team about the status of the fix and post here when I have an update.

**Comment by sfconcas** [comment]
@dmitridr Thank you so much, then I suppose the only thing left to do it's the option you just suggested, I am only afraid it'll delete my shop and it'll again start it all from zero. Will I have the same issue if I reconfigure the plugin? I mean, will I have to wait again for the shop to be processed again?..Or should I for now wait for your updates?

Sorry for the many questions

**Comment by dmitridr** [comment]
@sfconcas it depends on how much of a hurry you are in. If you want to have it fixed by today, then trying a reset may be your only way. You should not have the processing issue again, but processing may still take a few hours. If you are not in a rush, I recommend waiting. I think the shops team migration will finish before long, but still waiting on an update from them for how much longer.

**Comment by sfconcas** [comment]
@dmitridr 

and I am back to square one, after trying as you suggested the force re sync a few times nothing happened, so I just deleted and set it again as you suggested, now I don't have any more items to add in my website and facebook shop, so if you guys can keep an eye on the shop and see if it can be reviewed or fixed I truly appreciate it. 

I am sure you already have the details but just in case the facebook shop page is:

https://business.facebook.com/JSLaserCraft/shop/

Once again, thank you so much for the help, I don't know how we all could have done without your help!

**Comment by sfconcas** [comment]
@dmitridr that was faster than I thought, the shop has already been approved! can I ask you something? Variations, are they added through facebook? I have them on the website but not all of them are displayed on facebook, I guess fb uses them differently, not sure, I need to figure that out.... and would you know how can I use the shoppable instagram now that facebook is active? 

thank you so much, for everything, I will never say it enough

**Comment by FlairTrade** [comment]
Appreciate you keeping us updated. I reset my shop to resolve my issues, couldn’t wait this long. Time is money😜

**Comment by sfconcas** [comment]
@dmitridr Guys thank you very much for being so amazing, the shop is up and running, I don't thing I have nay issues now. I am not sure this is a bag or just the way fb import listings but I thought t'd ask before opening a ticket.I have variations in every listing, but instead of seeing variations, well I see, but in the dropdown menu of the variations I can only see "any" instead of the name of the variations, is that something normal or something that tI can fix?

Perhaps the screenshots will help,

Regards, 
Simon 
<img width="708" alt="screen shot 2018-06-09 at 09 37 38" src="https://user-images.githubusercontent.com/11273224/41189639-01d3c910-6bc9-11e8-8622-c5ad8b9fdb7f.png">
<img width="576" alt="screen shot 2018-06-09 at 09 37 49" src="https://user-images.githubusercontent.com/11273224/41189640-035825ba-6bc9-11e8-881d-3a4c0e4882eb.png">

**Comment by dmitridr** [comment]
All shops should be fixed now. Please comment here or open a new issue if you're still seeing out of stock problems.

@sfconcas Facebook attempts to mirror the product configuration from WooCommerce. If your WooCommerce configuration for the product variants is set to `any` for one of the variations, we will likewise copy that configuration onto FB. Likewise, if you manually configuration each of your variations to have a specific Font, Facebook should copy over the variations that you have configured. Hope that helps clarify. If you think this is a bug (your setting on WooCommerce is not `any` for the variations of this product), feel free to open a separate issue!

**Comment by Joshuagier** [comment]
Hi guys,

Is this issue resolved? Because I am mine is still having this stock issue :( it is really disruptiove.

Our online store is: https://hetopia.com

<img width="1067" alt="online shopping men accessories" src="https://user-images.githubusercontent.com/40202198/41287232-a5abf956-6e74-11e8-8608-020b05e7d3b6.png">
<img width="1172" alt="online shopping mens watch" src="https://user-images.githubusercontent.com/40202198/41287233-a83ec0c2-6e74-11e8-8304-eef0fc6837eb.png">


However the data that is uploaded is correct see last image

<img width="1893" alt="hetopia online men fashion accessories" src="https://user-images.githubusercontent.com/40202198/41287375-22511112-6e75-11e8-8f39-d46c5f9931a2.png">

**Comment by Joshuagier** [comment]
I tried force resync but it still does not work

**Comment by hotshotfiregirl** [comment]
I am having this issue as well. All items are showing "Out Of Stock" but my inventory in my catalog shows "100 in stock".

**Comment by MaRad1987** [comment]
I also have the same issue but with one webshop while at the same time other one is working okay.
It's not woocommerce, I'm uploading all products with .csv files.

Did this happen again maybe?

**Comment by KaterynaQ** [comment]
Version 1.9.15 having this issue. Spent 30 mins with Facebook support they told me the problem is in the plugin. I don't know what to do, tried resync, reinstall, update, 5 different devices and 10 browsers... All the items are synced correctly in the catalog but all of them out of stock in shop

**Comment by hitlerphung** [comment]
I also have the same problem, my web is : https://maxbuy.com.vn/ and my facebook shop is https://www.facebook.com/pg/maxbuy.jsc/shop/

**Comment by knitcollect** [comment]
MY uploaded products on facebook is showing out of stock despite they are all in stock in my data feed and the catalog recognise that, but on facebook frontend shop it shows out of stock. this my shop https://web.facebook.com/Knitcollect/. kindly look into it.

**Comment by knitcollect** [comment]
URGENT!!
my facebook shop page keeps displaying my products as out of stock! See here https://web.facebook.com/pg/Knitcollect/shop/?ref=page_internal, despite doing everything by the book as shown in my datafeed file and catalog. Kindly fix.

**Comment by hnmkmh** [comment]
My facebook shop page also suffers from this situation, I have tried many ways, both deleting all catalogs and re-creating but also.
I've tried using the Facebook xml example file but got the same error

My facebook shop page: https://www.facebook.com/pg/AIRSPACE.Vietnam/shop/?ref=page_internal

Kindly fix.
Thanks!

**Comment by marketjm** [comment]
Same here, tried a catalog creation through the pixel, facebook xml plugins, can't find a solution to the out of stock error. "The availability of this product is 'in stock' but its inventory is equal to 0. This is contradictory, please fix this in your data source."

**Comment by chvillanuevap** [comment]
Any updates on this? I'm having the same issue today.

**Comment by Lekshmanan-B** [comment]
After Sync products from woo-commerce to Facebook. The Facebook shop takes the products to review. After reviewing the products, the Facebook shop shows the product as (Out of Stock) and makes the inventory count as zero. Need to fix this error.
